Sources/MapboxMaps/Annotations/ViewAnnotationManager.swift

```diff
@@ -201,22 +264,40 @@ public final class ViewAnnotationManager {
             validate(view)
 
             view.translatesAutoresizingMaskIntoConstraints = true
-            view.frame = CGRect(
-                origin: position.leftTopCoordinate.point,
-                size: CGSize(width: CGFloat(position.width), height: CGFloat(position.height))
-            )
+            let oldFrame = view.frame
+            let newFrame = position.frame
+
+            view.frame = newFrame
+
+            if oldFrame != view.frame {
+                viewsWithUpdatedFrame.insert(view)
+            }
+            if view.isHidden {
+                viewsWithUpdatedVisibility.insert(view)
+            }
             view.isHidden = false
             expectedHiddenByView[view] = false
             visibleAnnotationIds.insert(position.identifier)
         }
 
+        defer {
+            assert(viewsWithUpdatedFrame.allSatisfy { !$0.isHidden })
+            notifyViewAnnotationObserversFrameDidChange(for: Array(viewsWithUpdatedFrame))
+        }
+
         let annotationsToHide = Set<String>(viewsById.keys).subtracting(visibleAnnotationIds)
+
         for id in annotationsToHide {
             guard let view = viewsById[id] else { fatalError() }
             validate(view)
+            if !view.isHidden {
+                viewsWithUpdatedVisibility.insert(view)
+            }
             view.isHidden = true
             expectedHiddenByView[view] = true
         }
+
+        notifyViewAnnotationObserversVisibilityDidChange(for: Array(viewsWithUpdatedVisibility))
```

Comment:
Can we make it `defer` as well for consistency?

Reply:
Updated